platform/android/detect.py

```diff
@@ -116,19 +116,14 @@ def configure(env):
 
     # Build type
 
-    if env["target"].startswith("release"):
-        if env["optimize"] == "speed":  # optimize for speed (default)
-            # `-O2` is more friendly to debuggers than `-O3`, leading to better crash backtraces
-            # when using `target=release_debug`.
-            opt = "-O3" if env["target"] == "release" else "-O2"
-            env.Append(CCFLAGS=[opt, "-fomit-frame-pointer"])
-        elif env["optimize"] == "size":  # optimize for size
-            env.Append(CCFLAGS=["-Oz"])
```

Comment:
A couple differences for Android:

- `optimize=size` used to be `-Oz`, now it's `-Os` like for all other platforms. `-Oz` is apparently a more aggressive size optimization flag, but for the Web build we have this info:
```
# -Os reduces file size by around 5 MiB over -O3. -Oz only saves about
# 100 KiB over -Os, which does not justify the negative impact on
# run-time performance.
```

- `target=debug` used to force no optimization (`-O0`), now it uses debug optimization (`-Og`) like other platforms, which is a subset of `-O1` which does not hinder debugging.